node/services/eventService.js

```diff
@@ -68,13 +69,21 @@ module.exports = {
             });
     },
 
-    getByID(req, res) {
-        return EventModel.findOne({ _id: req.params.event_id })
-            .then(eventDocument => res.status(200).send(eventDocument))
-            .catch(error => {
-                console.log(error);
-                res.status(error.status || 500).send(error);
-            });
+    async getByID(req, res) {
+        try {
+            const eventDocument = await EventModel
+                .findOne({ _id: req.params.event_id })
+                .exec()
+                .then(res => res.toObject())
+
+            const attendeesDetails = await UserModel.getAttendeeDetails(eventDocument.attendees)
+            const eventWithAttendeeDetails = { ...eventDocument, attendeesDetails }
+
+            res.status(200).send(eventWithAttendeeDetails)
+        } catch(err) {
+            console.log(err);
```

Comment:
We shouldn't need to `console.log` the error here if we throw the `res.status(...).send(...)`

Reply:
But it's also trivial so don't worry about it :)